Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp

```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
     // - Select the algorithm depending on the input parameters
     //   Initialize the corresponding coefficients over k space
     //   PML is not supported.
-    algorithm = std::unique_ptr<PsatdAlgorithmRZ>(
-        new PsatdAlgorithmRZ(k_space, dm, n_rz_azimuthal_modes, norder_z, nodal, dt));
+    if (v_galilean[2] == 0) {
+         // v_galilean is 0: use standard PSATD algorithm
+        algorithm = std::unique_ptr<PsatdAlgorithmRZ>(
+            new PsatdAlgorithmRZ(k_space, dm, n_rz_azimuthal_modes, norder_z, nodal, dt));
+    } else {
+        // Otherwise: use the Galilean algorithm
+        algorithm = std::unique_ptr<GalileanPsatdAlgorithmRZ>(
+            new GalileanPsatdAlgorithmRZ(k_space, dm, n_rz_azimuthal_modes, norder_z, nodal, v_galilean, dt));
+    }
 
     // - Initialize arrays for fields in spectral space + FFT plans
     field_data = SpectralFieldDataRZ(realspace_ba, k_space, dm,
                                      algorithm->getRequiredNumberOfFields(),
                                      n_rz_azimuthal_modes, lev);
 }
 
-
+/* \brief Transform the component `i_comp` of MultiFab `field_mf`
+ *  to spectral space, and store the corresponding result internally
+ *  (in the spectral field specified by `field_index`) */
 void
-SpectralSolverRZ::ForwardTransform (amrex::MultiFab const & field_mf,
-                                    int const field_index,
+SpectralSolverRZ::ForwardTransform (amrex::MultiFab const & field_mf, int const field_index,
                                     int const i_comp) {
     WARPX_PROFILE("SpectralSolverRZ::ForwardTransform");
     field_data.ForwardTransform(field_mf, field_index, i_comp);
-}
+};
```

Comment:
warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```